pwndbg/gdblib/symbol.py

```diff
@@ -84,19 +84,25 @@ def get(address: int, gdb_only=False) -> str:
                 res = pwndbg.ida.Name(address) or pwndbg.ida.GetFuncOffset(address)
                 return res or ""
 
-    # Expected format looks like this:
-    # main in section .text of /bin/bash
-    # main + 3 in section .text of /bin/bash
-    # system + 1 in section .text of /lib/x86_64-linux-gnu/libc.so.6
-    # No symbol matches system-1.
-    a, b, c, _ = result.split(maxsplit=3)
-
-    if b == "+":
-        return "%s+%s" % (a, c)
-    if b == "in":
-        return a
+    # See https://github.com/bminor/binutils-gdb/blob/d1702fea87aa62dff7de465464097dba63cc8c0f/gdb/printcmd.c#L1594-L1624
+    # The most often encountered formats looks like this:
+    #   "main in section .text of /bin/bash"
+    #   "main + 3 in section .text of /bin/bash"
+    #   "system + 1 in section .text of /lib/x86_64-linux-gnu/libc.so.6"
+    #   "No symbol matches system-1"
+    # But there are some others that we have to account for as well
+    if " in section " in result:
+        loc_string, _ = result.split(" in section ")
```

Comment:
@disconnect3d 
Somehow, I got an error: `context: too many values to unpack (expected 2) (<class 'ValueError'>` here.
I'm not sure how to reproduce it properly, but here's the binary(a warmup pwn from ACSC 2023) I trigger the error: [bug.zip](https://github.com/pwndbg/pwndbg/files/10895507/bug.zip)
The bug should be reproduced by `python solve.py` on Ubuntu 20.04 after this commit.
It’s caused by the multiple lines output like this:
```
main in section .text of /tmp/vaccine
main in section .text of /tmp/vaccine
```
(It’s really weird tho, not sure why it loads the remote binary and the local file at the same time.

----

Btw, maybe we should use `partition` instead of `split` here if we only want to return the first result?
```console
$ python -m timeit 'loc_string, _, _ = "a in section xxxx".partition("in section")'
2000000 loops, best of 5: 109 nsec per loop
$ python -m timeit 'loc_string, _ = "a in section xxxx".split("in section")'
2000000 loops, best of 5: 128 nsec per loop
```
The performance seems a little bit better. (but I’m not sure how it works behind actually :p )

Reply:
@lebr0nli can you check if `info symbol` returns the same if Pwndbg is not loaded? I wonder if that's Pwntools causing this or us.

Reply:
It seems this can be Pwntools since `gdb ./vaccine` and `info symbol main` with Pwndbg shows just one symbol.

Reply:
re: using `str.partition` - I agree we should probably use it, but I am still wondering how to fix this issue with newlines. We could take whatever first symbol appears in theory.

We could probably also do that with just: `loc_string, *_ = result.partition(...)`?

Reply:
@lebr0nli Woah. It seems this double symbol is caused by Pwndbg!

Reply:
This is caused by `_add_main_exe_to_symbols`. But this is still a bug: we should probably just first take symbol if multiple of them appears.

Reply:
Fwiw this is only happening with the combination with Pwntools. I think Pwntools may do sth like `file ./main` and load the file again this way...

Reply:
> Woah. It seems this double symbol is caused by Pwndbg!

Yeah, I can confirm that it only showed one line without pwndbg here.

> We could probably also do that with just: `loc_string, *_ = result.partition(...)`?

Yes, this looks better!

Reply:
> Yes, this looks better!

Ahh, wait, it looks better, but maybe doesn't work better xD

```console
$ python -m timeit 'loc_string, _, _ = "a in section xxxx".partition("in section")'
2000000 loops, best of 5: 119 nsec per loop
$ python -m timeit -n 2000000 'loc_string, *_ = "a in section xxxx".partition("in section")'
2000000 loops, best of 5: 228 nsec per loop
```

Reply:
Okay, I am wrong. This happens without Pwntools and can be reproduced by:
```
gdbserver :1234 ./vaccine
```
and then:
```
gdb --quiet --ex 'target remote :1234'
```

Then, `info symbol main` shows two symbols.